fcrepo-jms/src/test/java/org/fcrepo/integration/jms/observer/HeadersJMSIT.java

```diff
@@ -79,108 +105,74 @@
 
     private MessageConsumer consumer;
 
-    private volatile Set<Message> messages;
-
-    private JcrTools jcrTools = new JcrTools(true);
+    private volatile Set<Message> messages = new HashSet<>();
 
     private static final Logger LOGGER = getLogger(HeadersJMSIT.class);
 
-    /**
-     * Time to wait for a set of test messages.
-     */
-    private static final long TIMEOUT = 2000;
-
-    @Test
-    public void testIngestion() throws RepositoryException,
-                               InterruptedException, JMSException {
+    @Test(timeout = TIMEOUT)
+    public void testIngestion() throws RepositoryException {
 
-        final String pid = "/testIngestion";
-        final String expectedEventType =
-            REPOSITORY_NAMESPACE + EventType.valueOf(NODE_ADDED).toString();
-        LOGGER.debug("Expecting a {} event", expectedEventType);
-        Boolean success = false;
+        LOGGER.debug("Expecting a {} event", INGESTION_EVENT_TYPE);
 
         final Session session = repository.login();
         try {
-            final Node node = jcrTools.findOrCreateNode(session, pid);
-            node.addMixin(FEDORA_CONTAINER);
+            containerService.findOrCreate(session, testIngested);
             session.save();
-
-            final Long start = currentTimeMillis();
-            synchronized (this) {
-                while ((currentTimeMillis() - start < TIMEOUT) && (!success)) {
-                    for (final Message message : messages) {
-                        if (getIdentifier(message).equals(pid)) {
-                            if (getEventTypes(message).contains(expectedEventType)) {
-                                success = true;
-                            }
-                        }
-                    }
-                    LOGGER.debug("Waiting for next message...");
-                    wait(1000);
-                }
-            }
+            awaitMessageOrFail(testIngested, INGESTION_EVENT_TYPE);
         } finally {
             session.logout();
         }
-        assertTrue(
-                "Found no message with correct identifer and correct event type!",
-                success);
     }
 
-    @Test
-    public void testRemoval() throws RepositoryException, InterruptedException,
-                             JMSException {
+    private void awaitMessageOrFail(final String id, final String eventType) {
+        await().pollInterval(ONE_SECOND).until(new Callable<Boolean>() {
```

Comment:
If the `pollInterval` is 1sec and the timeout (which starts before the polling) is 2secs, it would seem that effectively the wait will just be a single second. Should we adjust either the polling interval or the timeout so the polling gets at least two tries?